Framework/Reflectometry/src/ReflectometryWorkflowBase2.cpp

```diff
@@ -681,37 +722,34 @@ std::string ReflectometryWorkflowBase2::findProcessingInstructions(const Instrum
 
   const std::string analysisMode = getProperty("AnalysisMode");
 
+  boost::optional<size_t> maybeStart;
+  boost::optional<size_t> maybeStop;
   if (analysisMode == "PointDetectorAnalysis") {
-    const std::vector<double> pointStart = instrument->getNumberParameter("PointDetectorStart");
-    const std::vector<double> pointStop = instrument->getNumberParameter("PointDetectorStop");
-
-    if (pointStart.empty() || pointStop.empty())
-      throw std::runtime_error("Could not find 'PointDetectorStart' and/or "
-                               "'PointDetectorStop' in parameter file. Please "
-                               "provide processing instructions manually or "
-                               "set analysis mode to 'MultiDetectorAnalysis'.");
-
-    const auto detStart = static_cast<int>(pointStart[0]);
-    const auto detStop = static_cast<int>(pointStop[0]);
-
-    auto instructions = std::to_string(detStart);
-    if (detStart != detStop)
-      instructions += ":" + std::to_string(detStop);
-    return instructions;
+    maybeStart = getDetectorParamOrNone(instrument, inputWS, "PointDetectorStart");
+    maybeStop = getDetectorParamOrNone(instrument, inputWS, "PointDetectorStop");
+    if (!maybeStart || !maybeStop) {
+      throw std::runtime_error(
+          "Could not find 'PointDetectorStart' and/or 'PointDetectorStop' in parameter file. Please provide processing "
+          "instructions manually or set analysis mode to 'MultiDetectorAnalysis'.");
+    }
+  } else {
+    maybeStart = getDetectorParamOrNone(instrument, inputWS, "MultiDetectorStart");
+    maybeStop = getDetectorParamOrNone(instrument, inputWS, "MultiDetectorStop");
+    if (!maybeStart) {
+      throw std::runtime_error("Could not find 'MultiDetectorStart in parameter file. Please provide processing "
```

Comment:
A think there is a missing single quotation in this line after `MultiDetectorStart`

Reply:
Good spot!